api-client/src/main/java/com/xing/api/RealCallSpec.java

```diff
@@ -231,9 +237,22 @@ public void cancel() {
         return this;
     }
 
+    @Override public CallSpec<RT, ET> timeouts(int connectTimeout, int readTimeout) {
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+        return this;
+    }
+
     /** Returns a raw {@link Call} pre-building the targeted request. */
     private Call createRawCall() {
-        return api.client().newCall(builder.request());
+        OkHttpClient client = api.client();
+        if (readTimeout != -1 && connectTimeout != -1) {
```

Comment:
I don't quite like this -1 checks here, can you please think of a nicer way than comparing both != -1. Also, might it be possible that only one of the 2 values is -1 if you just want to set the `readTimeout` for example?